presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpillerFactory.java

```diff
@@ -80,6 +90,29 @@ private static ListeningExecutorService createExecutorServiceOfSize(int nThreads
         return MoreExecutors.listeningDecorator(executorService);
     }
 
+    @PostConstruct
+    public void cleanupOldSpillFiles()
+    {
+        spillPaths.forEach(this::cleanupOldSpillFiles);
+    }
+
+    private void cleanupOldSpillFiles(Path path)
+    {
+        Arrays.stream(path.toFile().listFiles(new PatternFilenameFilter(SPILL_FILE_PATTERN))).forEach(
```

Comment:
If you are going to use the new Java 7 file system apis, I'd suggest using them all the way.  Here is a tutorial on listing files: https://docs.oracle.com/javase/tutorial/essential/io/dirs.html#listdir